### PR TITLE
Overview new ui

### PR DESCRIPTION
**Pull request by quacklabs**
What this PR does:
- resolves https://github.com/raedahgroup/dcrios/issues/506
- added listeners for realtime update of sync status
- sync restarts on network failure
- Sync status live reporting done
- Wallet online status reads network status from Reachability layer
- Floating Send/Receive button is not included as it appears across multiple views which makes it a feature on its own. 

**Overview before updates**

**Sync page before updates**

**AFTER UPDATES**

**overview: no transactions**

**Syncing: details shown (scrolled)**

**Syncing: Restarting Sync**

**Syncing: No details (progress bar only)**

Wallet not synced:

### CODE REVIEW

File: Decred Wallet/Extensions/UIStackView.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+//
+//  Created by Sprinthub on 05/09/2019.
+//  Copyright © 2019 Decred. All rights reserved.
+//
+
+import UIKit
+
+extension UIStackView {
+    
+    func cornerRadius(_ radius: CGFloat){
```
Comment (neddoz) [inline Decred Wallet/Extensions/UIStackView.swift]: There is a lot going on here apart from just setting the corner radius. You might want to extract a few things out of this or rename the method to something more descriptive of all it does.

**Comment by neddoz** [comment]
Please Counter check with the designs. For instance, the no transactions section of the overview section ought to be less in height if there are no transactions. Also on synced, there should be no more details button. It should just show the latest block and the number of peers. Currently, I'm seeing this.
Screenshot 2019-09-11 at 14 36 53

File: Decred Wallet/Extensions/UIStackView.swift (hunk 1 above)
Comment (neddoz) [reply Decred Wallet/Extensions/UIStackView.swift]: I also noticed in the designs the decred logo is fixed in the navbar and not part of the overview page. When scrolling it shouldn't scroll together with the view.

File: Decred Wallet.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1361,7 +1361,7 @@
 				B30A37CE228F3AF000F78629 /* Testnet Release */,
 			);
 			defaultConfigurationIsVisible = 0;
-			defaultConfigurationName = "Mainnet Release";
+			defaultConfigurationName = "Testnet Release";
```
[the remark this answers is not in the document]
Comment (quacklabs) [reply Decred Wallet.xcodeproj/project.pbxproj]: This happened cuz I accidentally clicked on a different scheme while trying to run. not necessary at all and changed in later commits.

File: Decred Wallet/Extensions/UIStackView.swift (hunk 1 above)
Comment (quacklabs) [reply Decred Wallet/Extensions/UIStackView.swift]: added comments to address observation 1 in e50fe14

File: Decred Wallet.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1077,7 +1077,7 @@
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIconTestnet;
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				CODE_SIGN_STYLE = Automatic;
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = 3VF3KBAZJF;
```
[the remark this answers is not in the document]
Comment (quacklabs) [reply Decred Wallet.xcodeproj/project.pbxproj]: resolved in e50fe14

File: Decred Wallet/Extensions/UIStackView.swift [hunk 4]
```diff
@@ -0,0 +1,26 @@
+//
+//  UIStackView.swift
+//  Decred Wallet
+//
+//  Created by Sprinthub on 05/09/2019.
```
[the remark this answers is not in the document]
Comment (quacklabs) [reply Decred Wallet/Extensions/UIStackView.swift]: resolved in e50fe14

File: Decred Wallet/Extensions/UIStackView.swift (hunk 1 above)
Comment (quacklabs) [reply Decred Wallet/Extensions/UIStackView.swift]: The fixed navbar with a fixed logo and dynamic text (switches from Overview to wallet amount and back to overview on scroll) is achievable when the nav is abstracted. this is being addressed here:  [https://github.com/quacklabs/dcrios/tree/navmenu-new-ui](url) as delegate methods to trigger the change will be added here as other views also modify the title label.

File: Decred Wallet/Features/Overview/OverviewViewController.swift [hunk 5]
```diff
@@ -7,152 +7,593 @@
...
+    func onSyncEndedWithError(_ error: String) {
+        self.syncing = false
+        self.syncStatusImage.image = UIImage(named: "icon-cancel")
+        self.syncStatusLabel.text = LocalizedStrings.walletNotSynced
+    }
+    
+    func debug(_ debugInfo: DcrlibwalletDebugInfo) {
+        
+    }
+
```
Comment (macsleven) [inline Decred Wallet/Features/Overview/OverviewViewController.swift]: Remove extra empty space in this file

File: Decred Wallet/Utils/LocalizedStrings.swift [hunk 6]
```diff
@@ -15,7 +15,6 @@ struct LocalizedStrings {
     /* Wallet Setup */
     static let createNewWallet = NSLocalizedString("createNewWallet", comment: "")
     static let restoreExistingWallet = NSLocalizedString("restoreExistingWallet", comment: "")
-    static let createOrRecoverWallet = NSLocalizedString("createOrRecoverWallet", comment: "")
```
[the remark this answers is not in the document]
Comment (itswisdomagain) [reply Decred Wallet/Utils/LocalizedStrings.swift]: This is already done in the PR it relates to: #512

File: Decred Wallet.xcodeproj/xcshareddata/xcschemes/Decred Wallet.xcscheme [hunk 7]
```diff
@@ -15,7 +15,7 @@
             <BuildableReference
                BuildableIdentifier = "primary"
                BlueprintIdentifier = "B3B4698E228F0C2D00A68EDD"
-               BuildableName = "Decred Wallet.app"
+               BuildableName = "Decred Wallet Testnet.app"
```
Comment (itswisdomagain) [inline Decred Wallet.xcodeproj/xcshareddata/xcschemes/Decred Wallet.xcscheme]: This is wrong.

File: Decred Wallet/en.lproj/Localizable.strings [hunk 8]
```diff
@@ -140,6 +143,21 @@
 "total"  = "total:";
 "stageTimes" = "Stage Times";
 "tapToViewInformation" = "Tap for more information";
+// refactor additions. some values above may not longer be used
```
[the remark this answers is not in the document]
Comment (quacklabs) [reply Decred Wallet/en.lproj/Localizable.strings]: resolved in 03100d1

File: Decred Wallet/en.lproj/Localizable.strings [hunk 9]
```diff
@@ -110,6 +111,9 @@
 "totalBalance" = "Total Balance";
 "restartingSynchronization" = "Restarting Synchronization";
 "startingSynchronization" = "Starting Synchronization";
+"fetchedHeaders" = "%d of ~%d ";
+"bestBlockAgebehind" = "%@ behind";
+
```
[the remark this answers is not in the document]
Comment (quacklabs) [reply Decred Wallet/en.lproj/Localizable.strings]: resolved in 03100d1

File: Decred Wallet/en.lproj/Localizable.strings [hunk 10]
```diff
@@ -140,6 +143,21 @@
...
+"connectedTo" = "Connected to %d peers";
+"latestBlockAge" = "Latest block %d% :- %@";
+"showDetails" = "Show details";
+"hideDetails" = "Hide details";
+"syncTimeLeft" = "%@ left";
+"syncSteps" = "Step %d%/3";
+"blockHeadersFetched" = "Block headers fetched";
+"connectedPeersCount" = "Connected peers count";
+"syncingProgress" = "Syncing progress";
+
```
[the remark this answers is not in the document]
Comment (quacklabs) [reply Decred Wallet/en.lproj/Localizable.strings]: resolved in 03100d1

File: Decred Wallet/Utils/LocalizedStrings.swift [hunk 11]
```diff
@@ -143,6 +144,21 @@ struct LocalizedStrings {
     static let total = NSLocalizedString("total", comment: "")
     static let stageTimes = NSLocalizedString("stageTimes", comment: "")
     static let tapToViewInformation = NSLocalizedString("tapToViewInformation", comment: "")
+    //some values above may no longer be used
```
[the remark this answers is not in the document]
Comment (quacklabs) [reply Decred Wallet/Utils/LocalizedStrings.swift]: resolved in eb5faf8

File: Decred Wallet/Features/Wallet Setup/WalletSetupViewController.swift [hunk 12]
```diff
@@ -10,28 +10,10 @@ import Foundation
 import UIKit
 
 class WalletSetupViewController: WalletSetupBaseViewController {
-    @IBOutlet weak var infoText: UILabel!
```
[the remark this answers is not in the document]
Comment (quacklabs) [reply Decred Wallet/Features/Wallet Setup/WalletSetupViewController.swift]: changes reverted in b55953a

**Pull request closed without merging**